patches/tModLoader/Terraria/ModLoader/DrawLayer.cs

```diff
@@ -1,280 +1,48 @@
-using System;
 using System.Collections.Generic;
 
 namespace Terraria.ModLoader
 {
-	/// <summary>
-	/// This class represents a layer of the drawing of an object, using a certain type of InfoType to help with its drawing.
-	/// </summary>
+	/// <summary> This class represents a layer of the drawing of an object, using a certain type of InfoType to help with its drawing. </summary>
 	/// <typeparam name="InfoType"></typeparam>
-	public abstract class DrawLayer<InfoType>
+	public abstract class DrawLayer<InfoType> : ModType
 	{
-		/// <summary>
-		/// The name of the mod to which this DrawLayer belongs.
-		/// </summary>
-		public readonly string mod;
-		/// <summary>
-		/// The name which identifies this DrawLayer.
-		/// </summary>
-		public readonly string Name;
-		/// <summary>
-		/// The parent of this DrawLayer. If the parent is not drawn, this layer will not be drawn either. Defaults to null, which skips the parent check.
-		/// </summary>
-		public readonly DrawLayer<InfoType> parent;
-		/// <summary>
-		/// The delegate of this method, which can either do the actual drawing or add draw data, depending on what kind of layer this is.
-		/// </summary>
-		public readonly Action<InfoType> layer;
-		/// <summary>
-		/// Whether or not this DrawLayer should be drawn. For vanilla layers, this will be set to true before all drawing-related hooks are called. For modded layers, you must set this to true or false yourself.
-		/// </summary>
+		/// <summary> The delegate used for draw layers. </summary>
+		public delegate void LayerFunction(ref InfoType info);
+
+		/// <summary> The parent of this DrawLayer. If the parent is not drawn, this layer will not be drawn either. Defaults to null, which skips the parent check.</summary>
+		public virtual DrawLayer<InfoType> Parent { get; }
+
+		/// <summary> Whether or not this DrawLayer should be drawn. For vanilla layers, this will be set to true before all drawing-related hooks are called. For modded layers, you must set this to true or false yourself. </summary>
 		public bool visible = true;
 
-		/// <summary>
-		/// Creates a DrawLayer with the given mod name, identifier name, and drawing action.
-		/// </summary>
-		/// <param name="mod"></param>
-		/// <param name="name"></param>
-		/// <param name="layer"></param>
-		protected DrawLayer(string mod, string name, Action<InfoType> layer) {
-			this.mod = mod;
-			this.Name = name;
-			this.parent = null;
-			this.layer = layer;
-		}
+		/// <summary> This layer's depth. Determines the order in which layers will appear. </summary>
+		public float depth;
```

Comment:
Please, if we can replace this with a `Before/After` system. Depths changing each frame is meh, but using fractional depths to arbitrarily inb4 is even more sad.